### PR TITLE
ReadDir add 'offset' param

### PR DESCRIPTION
**Pull request by Tricky511**

### CODE REVIEW

File: intercept/src/client.rs [hunk 1]
```diff
@@ -216,69 +215,76 @@ impl Client {
             return Ok(0);
```
Comment (luanshaotong) [inline intercept/src/client.rs:215]: Why not remove this?

File: src/server/storage_engine/default_engine.rs [hunk 2]
```diff
@@ -180,10 +211,58 @@ impl StorageEngine for DefaultEngine {
...
+                                if total + reclen as u32 > size {
+                                    break;
+                                }
+                                total += reclen as u32;
+                                unsafe {
+                                    (*(dirp_ptr as *mut dirent64)).d_ino = 1;
+                                    (*(dirp_ptr as *mut dirent64)).d_off = 2;
+                                    (*(dirp_ptr as *mut dirent64)).d_reclen = reclen;
+                                    (*(dirp_ptr as *mut dirent64)).d_type =
+                                        match value.1.as_bytes()[0] {
```
Comment (uran0sH) [inline src/server/storage_engine/default_engine.rs:234]: I think we don't need to use unsafe here.

File: src/server/storage_engine/default_engine.rs [hunk 3]
```diff
@@ -164,7 +166,36 @@ impl StorageEngine for DefaultEngine {
         }
     }
 
-    fn read_directory(&self, path: String) -> Result<Vec<u8>, EngineError> {
+    // fn read_directory(&self, path: String) -> Result<Vec<u8>, EngineError> {
```
Comment (luanshaotong) [inline src/server/storage_engine/default_engine.rs:169]: Pls remove the comment.

File: intercept/src/client.rs (hunk 1 above)
Comment (Tricky511) [reply intercept/src/client.rs]: removed

File: src/server/storage_engine/default_engine.rs (hunk 2 above)
Comment (Tricky511) [reply src/server/storage_engine/default_engine.rs]: How to do

File: src/server/storage_engine/default_engine.rs (hunk 3 above)
Comment (Tricky511) [reply src/server/storage_engine/default_engine.rs]: removed

File: intercept/src/client.rs (hunk 1 above)
Comment (luanshaotong) [reply intercept/src/client.rs]: Pls also remove it in fuse_client.rs.

Comment (Tricky511) [reply intercept/src/client.rs]: done

**Review by luanshaotong — APPROVED** [review]
LGTM

**Pull request merged**